app/shell2/code.cpp

```diff
@@ -133,7 +135,17 @@ FullCommandList::FullCommandList(EgalitoInterface *egalito) {
             {"-k", ArgumentSpec({"-k"}, ArgumentSpec::TYPE_FLAG)}
         }, {}, 0),
         [this, egalito] (ShellState &state, ArgumentValueList &args) {
-            std::string output = tempnam("/tmp", "ega-");
+            //std::string output = tempnam("/tmp", "ega-");
+            char pfnam[TPATH_MAX];
+            char tfnam[TPATH_MAX];
+
+            memset(pfnam, 0, TPATH_MAX);
+            memset(tfnam, 0, TPATH_MAX);
```

Comment:
`char pfnam[TPATH_MAX] = {};` since this is C++

Reply:
Given `memset`, I didn't zero-init the array, but sure that's better.